src/sensors/SensorBuilder.h

```diff
@@ -0,0 +1,494 @@
+		//	return SensorTypeID::LSM6DSO;
+		// }
+		// if (SoftFusionLSM6DSR::checkPresent(sensorID, imuAddress)) {
+		//	return SensorTypeID::LSM6DSR;
+		// }
+		// if (SoftFusionMPU6050::checkPresent(sensorID, imuAddress)) {
+		//	return SensorTypeID::MPU6050;
+		// }
+		if (SoftFusionICM45686::checkPresent(sensorID, imuAddress)) {
+			return SensorTypeID::ICM45686;
+		}
+		return BNO080Sensor::checkIfPresent(sensorID, imuAddress, intPin);
+		// if (SoftFusionICM45605::checkPresent(sensorID, imuAddress)) {
+		//	return SensorTypeID::ICM45605;
+		// }
+
+		return SensorTypeID::Unknown;
+	}
+
+	SensorTypeID findSensorType(
+		uint8_t sensorID,
+		RegisterInterface& imuInterface,
+		float rotation,
+		SensorInterface* sensorInterface,
+		bool optional,
+		PinInterface* intPin,
+		int extraParam
+	) {
+		sensorInterface->init();
+		sensorInterface->swapIn();
+		// if (SoftFusionLSM6DS3TRC::checkPresent(sensorID, imuInterface))
+		// { 	return SensorTypeID::LSM6DS3TRC;
+		// }
+		// if (SoftFusionICM42688::checkPresent(sensorID, imuInterface)) {
+		//	return SensorTypeID::ICM42688;
+		//}
+		if (SoftFusionBMI270::checkPresent(sensorID, imuInterface)) {
+			return SensorTypeID::BMI270;
+		}
+		if (SoftFusionLSM6DSV::checkPresent(sensorID, imuInterface)) {
+			return SensorTypeID::LSM6DSV;
+		}
+		// if (SoftFusionLSM6DSO::checkPresent(sensorID, imuInterface)) {
+		//	return SensorTypeID::LSM6DSO;
+		// }
+		// if (SoftFusionLSM6DSR::checkPresent(sensorID, imuInterface)) {
+		//	return SensorTypeID::LSM6DSR;
+		// }
+		// if (SoftFusionMPU6050::checkPresent(sensorID, imuInterface)) {
+		//	return SensorTypeID::MPU6050;
+		// }
+		if (SoftFusionICM45686::checkPresent(sensorID, imuInterface)) {
+			return SensorTypeID::ICM45686;
+		}
+		return BNO080Sensor::checkIfPresent(sensorID, sensorInterface, intPin);
+		// if (SoftFusionICM45605::checkPresent(sensorID, imuInterface)) {
+		//	return SensorTypeID::ICM45605;
+		// }
+
+		return SensorTypeID::Unknown;
+	}
+
+	template <typename ImuType>
+	std::unique_ptr<::Sensor> buildSensor(
+		uint8_t sensorID,
+		RegisterInterface& imuInterface,
+		float rotation,
+		SensorInterface* sensorInterface,
+		bool optional = false,
+		PinInterface* intPin = nullptr,
+		int extraParam = 0
+	) {
+		return buildSensorReal<ImuType>(
+			sensorID,
+			imuInterface,
+			rotation,
+			sensorInterface,
+			optional,
+			intPin,
+			extraParam
+		);
+	}
+
+	template <typename ImuType>
+	std::unique_ptr<::Sensor> buildSensor(
+		uint8_t sensorID,
+		uint8_t imuAddress,
+		float rotation,
+		SensorInterface* sensorInterface,
+		bool optional = false,
+		PinInterface* intPin = nullptr,
+		int extraParam = 0
+	) {
+		uint8_t address = imuAddress > 0 ? imuAddress : ImuType::Address + sensorID;
+		return buildSensorReal<ImuType>(
+			sensorID,
+			*(new I2CImpl(address)),
+			rotation,
+			sensorInterface,
+			optional,
+			intPin,
+			extraParam
+		);
+	}
+
+	template <typename ImuType>
+	std::unique_ptr<::Sensor> buildSensorReal(
```

Comment:
I'm not entirely sure why breaking it into a function and functionReal is entirely necessary